src/lightgbm/src/main/scala/TrainParams.scala

```diff
@@ -17,12 +17,14 @@ abstract class TrainParams extends Serializable {
   def featureFraction: Double
   def maxDepth: Int
   def minSumHessianInLeaf: Double
+  def numMachines: Int
 
   override def toString(): String = {
     s"is_pre_partition=True boosting_type=gbdt tree_learner=$parallelism num_iterations=$numIterations " +
       s"learning_rate=$learningRate num_leaves=$numLeaves " +
       s"max_bin=$maxBin bagging_fraction=$baggingFraction bagging_freq=$baggingFreq bagging_seed=$baggingSeed " +
-      s"feature_fraction=$featureFraction max_depth=$maxDepth min_sum_hessian_in_leaf=$minSumHessianInLeaf"
+      s"feature_fraction=$featureFraction max_depth=$maxDepth min_sum_hessian_in_leaf=$minSumHessianInLeaf " +
+      s"numMachines=$numMachines"
```

Comment:
@ywskycn i think the param should be `num_machines`. When using your version I was getting:

```
[LightGBM] [Warning] Unknown parameter: numMachines
```

after changing it to the `num_machines` the warning was gone and the training started to be synchronized across workers. 

The problem is that the training is very slow now it takes around 30 seconds per iteration. So I am now waiting for the results.

Reply:
yes, it can be num_machine or num_machines (see aliases), good catch:

https://github.com/Microsoft/LightGBM/blob/0584065c036414f6e4ae01e72314f2f9cbed7329/docs/Parameters.rst#network-parameters

num_machines, default = 1, type = int, aliases: num_machine, constraints: num_machines > 0

    the number of machines for parallel learning application
    this parameter is needed to be set in both socket and mpi versions